### ISSUE TITLE
Switch JSON serializer usage over to new source generator APIs

### ISSUE THREAD

**Issue by CoffeeFlux** [dotnet/aspnetcore#31877]
Until this happens, the default Blazor template will have a size regression: https://github.com/dotnet/runtime/issues/51311

New APIs: https://github.com/dotnet/runtime/issues/45448#issue-754827678

Initial attempt: https://github.com/eerhardt/aspnetcore/commits/json-tmp

cc: @mkArtakMSFT @eerhardt @layomia

**Comment by mkArtakMSFT** [comment]
@MackinnonBuck what is left here to be done?

**Comment by MackinnonBuck** [comment]
As of https://github.com/dotnet/aspnetcore/pull/54956, we've started to use the STJ source generator during WebAssembly startup. This was a targeted change whose goal was to improve startup performance. We also scoped the changes to be limited to those that didn't require new public API.

In the future, we could revisit this area by creating new public APIs that enable customers to utilize the STJ source generator for JS interop, persistent component state, etc. We currently have https://github.com/dotnet/aspnetcore/issues/39839 and https://github.com/dotnet/aspnetcore/issues/40683 to track specific scenarios that could benefit from supporting the STJ source generator, and we can open new issues in the future for additional scenarios.
